elements/pl-file-editor/pl-file-editor.js

```diff
@@ -100,6 +108,56 @@ window.PLFileEditor.prototype.updatePreview = function (html_contents) {
   }
 };
 
+window.PLFileEditor.prototype.initSettingsButton = function (uuid) {
+  var that = this;
+
+  this.settingsButton.click(function () {
+    ace.require(['ace/ext/themelist'], function (themeList) {
+      var themeSelect = that.modal.find('#' + uuid + '-themes');
+      themeSelect.empty();
+      for (const entries in themeList.themesByName) {
+        var caption = themeList.themesByName[entries].caption;
+        var theme = themeList.themesByName[entries].theme;
+
+        if (localStorage.getItem('pl-file-editor-theme') === theme) {
+          themeSelect.append(
+            $('<option>', {
+              value: theme,
+              text: caption,
+              selected: true,
+            })
+          );
+        } else {
+          themeSelect.append(
+            $('<option>', {
+              value: theme,
+              text: caption,
+            })
+          );
+        }
+      }
+    });
+    that.modal.modal('show');
+    // listen for changes to the theme select
+    that.modal.find('#' + uuid + '-themes').change(function () {
+      var theme = $(this).val();
+      that.editor.setTheme(theme);
+      console.log('theme', theme);
+    });
+  });
+
+  this.saveSettingsButton.click(function () {
+    var theme = that.modal.find('#' + uuid + '-themes').val();
+    that.editor.setTheme(theme);
+    localStorage.setItem('pl-file-editor-theme', theme);
+    that.modal.modal('hide');
+  });
+
+  this.closeSettingsButton.click(function () {
+    that.editor.setTheme(localStorage.getItem('pl-file-editor-theme'));
```

Comment:
This value isn't guaranteed to be present in local storage if the user never changed the theme, correct? We should be reverting back to whatever the theme was before the modal was opened, not necessarily what was in local storage.